feat: update referral program for new requirements (#1914)
vegaprotocol/specs: protocol/0083-RFPR-on_chain_referral_program.md

Providing a party has been associated with a referral set for long enough, they will become eligible for greater benefits as their referral sets running taker volume increases.

> Referrers will also the ability to increase the proportion of referee taker fees they receive by staking governance tokens ($VEGA).

Reviewer: I think this is missing the been "have"

To create an emphasis on community, collaboration, and competition. Referrers will be able to designate their referral set as a team. Teams will have additional fields which allow them to be visible on leaderboards and later to compete for team based rewards.

> Note, if a party wants to contribute towards a different referral set or compete as a member of a different team, a party is able to move between referral sets by "reapplying" a new referral code. However, if they do, their referral rewards will still be paid out to the referrer of the original referral set.

Reviewer 2: Shouldn't this also be that their referral rewards will also still be calculated based on the volume of their original referral set?

I am wondering here if this does mean people can need to be in two sets: their original refferer one for referrals and their current team for incentives, else they stop contributing to their original referrer's volume and maybe that's not ideal? 🤔

Author: > Shouldn't this also be that their referral rewards will also still be calculated based on the volume of their original referral set?

Yes this is true and is described in the "Setting benefit factors" section. This paragraph should be making that clear though so will edit.

```suggestion
Note, if a party wants to compete as a member of a different team, a party is able to move between teams by "reapplying" a new referral code. However, if they do, their referral rewards will still be paid out to the referrer of the original referral set and they will still contribute to the original referral sets volume.
```

Reviewer: We lack clarity on the naming here. I believe we should separate the concept of team and referral set in here for the sake of clarity.

Basically, you can change team, but not referral set. So this paragraph would be:
```suggestion
Note, if a referee wants to compete as a member of a different team, he is able to move between teams by "reapplying" a referral code. However, their referral rewards will still be computed based their original referral set performance and, thus, paid out to the their original referrer.
```

Reviewer 2: Agree that makes sense with this change

Enabling or changing the terms of the on-chain referral program can be proposed via governance. As part of the proposal, the proposer specifies the following fields:

- `benefit_tiers`: a list of dictionaries (with the below fields) defining the reward and discount factors from referrals
  - `minimum_running_notional_taker_volume`: the required [`referral_set_running_notional_taker_volume`](#referral-set-volumes) in quantum units for parties to access this tier

>   - `minimum_epochs`: the required number of epochs a party must have been in a referral set to access this tier

Reviewer: Should we replace party by referee since it only affect the referees? 
> the required number of epochs a referee must have been in a referral set to access this tier
